src/bytecode/concrete.py

```diff
@@ -421,6 +469,60 @@ def _remove_extended_args(
         if extended_arg is not None:
             raise ValueError("EXTENDED_ARG at the end of the code")
 
+    # Taken and adapted from exception_handling_notes.txt in cpython/Objects
+    @staticmethod
+    def _parse_varint(except_table_iterator: Iterator[int]) -> int:
+        b = next(except_table_iterator)
+        val = b & 63
+        while b & 64:
+            val <<= 6
+            b = next(except_table_iterator)
+            val |= b & 63
+        return val
+
+    def _parse_exception_table(
+        self, exception_table: bytes
+    ) -> List[ExceptionTableEntry]:
+        assert sys.version_info >= (3, 11)
+        table = []
+        iterator = iter(exception_table)
+        try:
+            while True:
+                start = self._parse_varint(iterator)
+                length = self._parse_varint(iterator)
+                end = start + length - 1  # Present as inclusive
+                target = self._parse_varint(iterator)
+                dl = self._parse_varint(iterator)
+                depth = dl >> 1
+                lasti = bool(dl & 1)
+                table.append(ExceptionTableEntry(start, end, target, depth, lasti))
+        except StopIteration:
+            return table
+
+    @staticmethod
+    def _encode_varint(value: int, set_begin_marker: bool = False) -> Iterator[int]:
+        # Encode value as a varint on 7 bits (MSB should come first) and set
+        # the begin marker if requested.
+        temp: List[int] = []
+        while value:
+            temp.append(value & 63 | (64 if temp else 0))
```

Comment:
Perhaps the `if ... else` branching can be avoided by appending the first value and shifting _before_ the loop